Modules/_io/winconsoleio.c

```diff
@@ -1000,6 +1001,9 @@ _io__WindowsConsoleIO_write_impl(winconsoleio *self, Py_buffer *b)
     wlen = MultiByteToWideChar(CP_UTF8, 0, b->buf, len, wbuf, wlen);
     if (wlen) {
         res = WriteConsoleW(self->handle, wbuf, wlen, &n, NULL);
+        if (!res)
```

Comment:
There's no point in calculating `len` if the call failed, so why not instead add this to the following test, i.e. `(res && n < wlen)`?